src/otx/core/model/entity/action_detection.py

```diff
@@ -0,0 +1,141 @@
+            @property
+            def device(self) -> device:
+                try:
+                    buf = next(self.buffers())
+                except StopIteration:
+                    return super().device
+                else:
+                    return buf.device
+
+        return build_mm_model(self.config, MODELS, self.load_from)
+
+    def _customize_inputs(self, entity: ActionDetBatchDataEntity) -> dict[str, Any]:
+        """Convert ActionClsBatchDataEntity into mmaction model's input."""
+        from mmaction.structures import ActionDataSample
+        from mmengine.structures import InstanceData
+
+        mmaction_inputs: dict[str, Any] = {}
+
+        mmaction_inputs["inputs"] = entity.images
+        entity_proposals = entity.proposals if entity.proposals else [None] * len(entity.bboxes)
+        mmaction_inputs["data_samples"] = [
+            ActionDataSample(
+                metainfo={
+                    "img_id": img_info.img_idx,
+                    "img_shape": img_info.img_shape,
+                    "ori_shape": img_info.ori_shape,
+                    "pad_shape": img_info.pad_shape,
+                    "scale_factor": img_info.scale_factor,
+                },
+                gt_instances=InstanceData(
+                    bboxes=bboxes,
+                    labels=labels,
+                ),
+                proposals=InstanceData(bboxes=proposals) if proposals is not None else None,
+            )
+            for img_info, bboxes, labels, proposals in zip(
+                entity.imgs_info,
+                entity.bboxes,
+                entity.labels,
+                entity_proposals,
+            )
+        ]
+
+        mmaction_inputs = self.model.data_preprocessor(data=mmaction_inputs, training=self.training)
+        mmaction_inputs["mode"] = "loss" if self.training else "predict"
+        return mmaction_inputs
+
+    def _customize_outputs(
+        self,
+        outputs: Any,  # noqa: ANN401
+        inputs: ActionDetBatchDataEntity,
+    ) -> ActionDetBatchPredEntity | OTXBatchLossEntity:
+        from mmaction.structures import ActionDataSample
+
+        if self.training:
+            if not isinstance(outputs, dict):
+                raise TypeError(outputs)
+
+            losses = OTXBatchLossEntity()
+            for k, v in outputs.items():
+                losses[k] = v
+            return losses
+
+        scores = []
+        labels = []
+        bboxes = []
+
+        for output in outputs:
+            if not isinstance(output, ActionDataSample):
+                raise TypeError(output)
+
+            output_scores, output_labels = output.pred_instances.scores.max(-1)
+            output.pred_instances.bboxes[:, 0::2] *= output.img_shape[1]
+            output.pred_instances.bboxes[:, 1::2] *= output.img_shape[0]
+            scores.append(output_scores)
+            bboxes.append(
+                tv_tensors.BoundingBoxes(
```

Comment:
Please use `tv_tensors.wrap()` to follow the best practices

https://pytorch.org/vision/0.16/auto_examples/transforms/plot_tv_tensors.html#using-tv-tensors-wrap

Reply:
I'm not sure we can use that function here. tv_tensros.wrap() requires torch tensor and reference tv_tensors, but here we only have torch tensors. Could you explain how can we use wrap function here?

Reply:
Sorry it was my miss